### PR TITLE
Added REST API tests [**/tasks GET, POST]

### PR DESCRIPTION
**Pull request by bsekachev**
### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added a description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2022 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: tests/rest_api/test_0004_tasks.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+    # [organization] supervisor can create task in project even he has no ownerships in this project
+    # TODO: SUPERVISOR DOES NOT SEE CREATED PROJECTS AND CANNOT CREATE TASKS
+
+    # def test_supervisor_can_create_task_in_project_without_ownership(self, users):
+    #     spec = {
+    #         'name': 'test supervisor can create task in project',
+    #         'project_id': self._ORG_PROJ_ID,
+    #     }
+    #     self._test_can_create_task_in_project(self._ORG_SUPERVISOR_NOT_OWNER, spec, org=self._ORG)
```
Comment (bsekachev) [inline tests/rest_api/test_0004_tasks.py:159]: @kirill-sizov Please, check the specification

File: tests/rest_api/test_0004_tasks.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    #     spec = {
+    #         'name': 'test worker with ownerships can create task in project',
+    #         'project_id': self._ORG_PROJ_ID,
+    #     }
+
+    #     worker3 = list(filter(lambda x: x['username'] == 'worker3', users))[0]
+
+    #     self._test_cannot_create_task_in_project(worker3['username'], spec, org=self._ORG)
+    #     self._assign_user_to_proj(worker3, self._ORG_PROJ_ID, org=self._ORG)
+    #     self._test_can_create_task_in_project(worker3['username'], spec, org=self._ORG)
```
Comment (bsekachev) [inline tests/rest_api/test_0004_tasks.py:181]: @kirill-sizov Please, check the specification

File: tests/rest_api/test_0004_tasks.py (hunk 1 above)
Comment (kirill-sizov) [reply tests/rest_api/test_0004_tasks.py:159]: `worker3` has privilege `worker`, but `worker` (as privilege) cannot create a task in this case. To resolve this problem need to change DB (for example change `user3` role from `supervisor` to `worker`) and after that use `user3` in these tests.

File: tests/rest_api/test_0004_tasks.py [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+class TestPostTasks:
+    _ADMIN = 'admin1'
+    _SAND_PROJ_ID = 1
+    _SAND_PROJ_OWNER = 'business1'
+    _SAND_PROJ_NOT_OWNER = 'business2'
+    _ORG = 'org2'
+    _ORG_PROJ_ID = 2
+    _ORG_SUPERVISOR_NOT_OWNER = 'user3'
+    _ORG_WORKER_NO_OWNERSHIP = 'worker3'
```
Comment (kirill-sizov) [inline tests/rest_api/test_0004_tasks.py:106]: Consider getting rid of hardcoded names and ids in tests. Avoiding hard-coded usernames, identifiers, etc. makes it easier to make new changes to the test database.

Fixture `find_user` that allow easy to find needed user with specific organization, role, and privilege:
```
user0 = find_users(org=1, role='maintainer')[0]['username']
user1 = find_users(privilege='user', org=2)[0]['username']
```

[Here](https://github.com/openvinotoolkit/cvat/blob/377fe2cfabf6d3f88f74afa9a49338253ec75543/tests/rest_api/conftest.py#L135-L141) possible solution (not merged yet) for searching needed user with specific ownership for the project

File: tests/rest_api/test_0004_tasks.py (hunk 1 above)
Comment (kirill-sizov) [reply tests/rest_api/test_0004_tasks.py:159]: Yes, I was wrong this comment belongs to [this](https://github.com/openvinotoolkit/cvat/blob/2cc26a30e42ed1ba3bf4b4870f18e1cc80ee274a/tests/rest_api/test_0004_tasks.py#L169-L181) test

File: tests/rest_api/test_0004_tasks.py (hunk 3 above)
Comment (bsekachev) [reply tests/rest_api/test_0004_tasks.py]: @kirill-sizov Updated. Now the PR depends on your PR #4351

**Comment by bsekachev** [comment]
@kirill-sizov Could you please look at the tests? I see there are some snyk issues, but I didn't install any 3rdparty components.

**Review by nmanovic — APPROVED** [review]
LGTM, I will merge.

**Pull request merged**